### PR TITLE
[WIP] Add a UI to create group and so  on

### PR DESCRIPTION
**Pull request by chx**
Even if we decide to use some other UI, I am learning og by porting this relatively simple routine. Without this I can't demo what works and what doesn't as I can't even create a group.

The submit routine is not yet ported but the form is. 

There's some overlap with https://github.com/amitaibu/og/pull/38/files#diff-0efd2133be56c76fae5a1a5f77bbb45bR206 but I am talking to with @damiankloip and we will figure out who keeps which version.

### CODE REVIEW

**Comment by amitaibu** [comment]
Awesome, thanks for starting this!

File: og_ui/og_ui.info.yml [hunk 1]
```diff
@@ -7,5 +7,6 @@ core: 8.x
 
 dependencies:
   - og
+  - field_ui
```
Comment (amitaibu) [inline og_ui/og_ui.info.yml]: I know it's a WIP, just some general remarks - OG UI shouldn't have a hard dependency on field_ui.

In fact, maybe in this port we will split OG UI into `OG UI` (for members/ groups admin) and `OG Admin` (for the site admin)

Comment (chx) [inline og_ui/og_ui.info.yml]: og_ui D7 had ` $url = array('!url' => l(t('Manage fields'), 'admin/structure/types/manage/' . str_replace('_', '-', $node_type) . '/fields'));` which basically created a soft dependency on Field UI. In D8 that's `$url = Link::createFromRoute(t('Manage fields'), FieldUI::getOverviewRouteInfo($entity_type_id, $bundle));` which is a hard dependency and that's why I added it.

Comment (amitaibu) [inline og_ui/og_ui.info.yml]: Yeah, I wasn't saying you did a mistake, I was mentioning what my goal is :)

File: og_ui/og_ui.module [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+use Drupal\og\Og;
+
+/**
+ * Implements hook_form_alter().
+ * @param $form
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ * @param $form_id
+ */
+function og_ui_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if ($form_state->getBuildInfo()['callback_object'] instanceof BundleEntityFormBase) {
```
Comment (timplunkett) [inline og_ui/og_ui.module]: `if ($entity_form_object = $form_state->getFormObject() instanceof BundleEntityFormBase) {`

File: src/Og.php [hunk 3]
```diff
@@ -195,13 +196,13 @@ public static function removeGroup($entity_type_id, $bundle_id) {
   /**
    * Return TRUE if field is a group audience type.
    *
-   * @param $field_config
+   * @param $field_definition
```
Comment (andypost) [inline src/Og.php]: should be `\Drupal\Core\Field\FieldDefinitionInterface`

**Comment by pfrenssen** [comment]
The `FieldAccessTest` that was just added to the 8.x-1.x fails. It's using `Og::createField()` whose signature is changed in this branch.

**Comment by damiankloip** [comment]
Needs a rebase.

**Pull request closed without merging**